activesupport/lib/active_support/core_ext/object/json.rb

```diff
@@ -26,22 +25,27 @@
 # bypassed completely. This means that as_json won't be invoked and the JSON gem will simply
 # ignore any options it does not natively understand. This also means that ::JSON.{generate,dump}
 # should give exactly the same results with or without active support.
-[Object, Array, FalseClass, Float, Hash, Integer, NilClass, String, TrueClass, Enumerable].each do |klass|
-  klass.class_eval do
-    def to_json_with_active_support_encoder(options = nil)
-      if options.is_a?(::JSON::State)
-        # Called from JSON.{generate,dump}, forward it to JSON gem's to_json
-        self.to_json_without_active_support_encoder(options)
-      else
-        # to_json is being invoked directly, use ActiveSupport's encoder
-        ActiveSupport::JSON.encode(self, options)
+
+module ActiveSupport
+  module CoreExt
+    module ToJsonWithActiveSupportEncoder
```

Comment:
@kirs @rafaelfranca I should've mentioned this too. Which namespace is more ideal, `ActiveSupport::CoreExt::...` or just `ActiveSupport::...`?